src/axom/spin/tests/spin_implicit_grid.cpp

```diff
@@ -833,32 +855,36 @@ TYPED_TEST(ImplicitGridExecTest, get_candidates_box_vectorized)
       {
         for(int k = 0; k < k_max; ++k)
         {
-          SpacePt queryPt {i * .1 + .05, j * .1 + .05, k * .1 + .05};
-          EXPECT_EQ(DIM, grid.getCandidates(queryPt).count());
+          queryPt[0] = SpacePt {i * .1 + .05, j * .1 + .05, k * .1 + .05};
+          EXPECT_EQ(DIM, grid.getCandidates(queryPt[0]).count());
         }
       }
     }
   }
 
-  //// Run some queries
-  BBox queryBoxes[] = {// Empty box -- covers no objects
-                       BBox {},
-                       // Box covers entire domain -- covers all objects
-                       bbox,
-                       // Box is larger than domain -- covers all objects
-                       BBox {SpacePt(-1.), SpacePt(2.)},
-                       // Box only covers first quadrant/octant of domain
-                       BBox {SpacePt(-1.), SpacePt(0.45)},
-                       // Box covers a single cell
-                       BBox {SpacePt(0.525), SpacePt(0.575)},
-                       // Box only covers last quadrant/octant of domain
-                       BBox {SpacePt(0.55), SpacePt(2.)},
-                       // Box covers middle of domain
-                       BBox {SpacePt(0.25), SpacePt(0.75)},
-                       // Box is inverted -- BoundingBox constructor fixes this
-                       BBox {SpacePt(2.), SpacePt(-1)}};
+  axom::deallocate(BBoxes);
+  axom::deallocate(queryPt);
 
+  //// Run some queries
   constexpr int N_QUERIES = 8;
+  BBox* queryBoxes = axom::allocate<BBox>(N_QUERIES, kernelAllocID);
```

Comment:
Would it make sense to switch the tests in this file over to `axom::Array`
(perhaps this was suggested by others as well?)